pkg/scheduler/framework/interface.go

```diff
@@ -385,6 +385,8 @@ type PreFilterPlugin interface {
 	// plugins must return success or the pod will be rejected. PreFilter could optionally
 	// return a PreFilterResult to influence which nodes to evaluate downstream. This is useful
 	// for cases where it is possible to determine the subset of nodes to process in O(1) time.
+	// When it returns Skip status, returned PreFilterResult and other fields in status are just ignored,
+	// and coupled Filter plugin will be skipped in this scheduling cycle.
 	PreFilter(ctx context.Context, state *CycleState, p *v1.Pod) (*PreFilterResult, *Status)
 	// PreFilterExtensions returns a PreFilterExtensions interface if the plugin implements one,
```

Comment:
Actually, another thought:

Should we explicitly skip calling PreFilterExtensions when PreFilter returned Skip?

Reply:
> Should we explicitly skip calling PreFilterExtensions when PreFilter returned Skip?

Good catch.. Yes, I believe we should skip. 

- we only use PreFilterExtensions in `SelectVictimsOnNode ` func in the default_preemption.
  - https://github.com/kubernetes/kubernetes/blob/2d534e4bea7c7ef412a03c921867d9027f6e489f/pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go#L144-L161
- The `SelectVictimsOnNode` run Filter plugins via `RunFilterPluginsWithNominatedPods` func.
  - https://github.com/kubernetes/kubernetes/blob/2d534e4bea7c7ef412a03c921867d9027f6e489f/pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go#L186
- The `RunFilterPluginsWithNominatedPods` func calls `RunFilterPlugins` to run Filter plugins.
  - https://github.com/kubernetes/kubernetes/blob/2d534e4bea7c7ef412a03c921867d9027f6e489f/pkg/scheduler/framework/runtime/framework.go#L836
- In `RunFilterPlugins`, Filter plugins in `SkipFilterPlugins` of CycleState are skipped.


So, it doesn't make sense to call `PreFilterExtensions.AddPod/RemovePod` of `SkipFilterPlugins` in the default_preemption because those plugins aren't executed after that.
Let me fix.

Reply:
> we only use PreFilterExtensions in SelectVictimsOnNode  func in the default_preemption.

It's wrong, we also use `PreFilterExtensions().AddPod()` in `addNominatedPods` as well, but the conclusion is the same since `addNominatedPods` is called only in `RunFilterPluginsWithNominatedPods`.
https://github.com/kubernetes/kubernetes/blob/2d534e4bea7c7ef412a03c921867d9027f6e489f/pkg/scheduler/framework/runtime/framework.go#L863